packages/interface/src/interface.ts

```diff
@@ -0,0 +1,165 @@
+/**
+ * @license
+ * Copyright (c) 2020 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+interface PolyfillsWindow extends Window {
+  ShadyCSS?: ShadyCSS;
+}
+
+interface ShadyCSS {
```

Comment:
would we be able to import these types if shadycss was in TypeScript?

Reply:
Only if this package had a dependency on the shady-css package

Reply:
> LGTM, but I'd like to talk just once more about what polyfills we might facade like this and whether this should be in shady-css... just to make sure we have it right before publishing

Yeah, I could be convinced either way.

Pros:

1. Maybe more logical organization. These are proxies for shadycss methods, so seems reasonable to expect that shadycss itself would provide it?

2. If we added more proxies for other WC polyfill methods here (e.g. shadydom), then some users will improt more code than they need. Many bundlers don't tree-shake within a module (though we're talking about tiny amounts of code either way). Also organizationally seems to make sense for each polyfill to provide its own functions (though the shadycss/shadydom distinction isn't all that meaningful to most users, I feel).

3. One less package for us to maintain and users to know about.

Cons:

1. NPM dependency bloat. It makes sense for libraries/components (e.g. lit-element, any re-usable component that e.g. sets a css property imperatively) to call these APIs by installing this dep, but they don't actually need the polyfill itself. No runtime cost of course, just install cost.

2. The optics of simplistic dependency analysis.

WDYT?

Reply:
I totally agree with your pros and cons. I'm just not 100% the conclusion I draw from them :)

Reply:
I'm leaning shadycss package, seems less weird.

Reply:
IMO it's better for the types to be defined once, and for us to do `import type` here

Reply:
Sorry, I messed up somehow with this thread by replying in the wrong place. My first reply, and the next two comments, don't have anything to do with your original question and Justin's first reply. Interested in your opinion about where this package should live, not sure if your last comment is related to that.